cmake/OpenCVFindLibsGUI.cmake

```diff
@@ -12,16 +12,28 @@ if(WITH_WIN32UI)
 endif()
 
 # --- QT4 ---
-ocv_clear_vars(HAVE_QT HAVE_QT5)
+ocv_clear_vars(HAVE_QT HAVE_QT5 HAVE_QT6)
 if(WITH_QT)
   if(NOT WITH_QT EQUAL 4)
-    find_package(Qt5 COMPONENTS Core Gui Widgets Test Concurrent REQUIRED NO_MODULE)
-    if(Qt5_FOUND)
-      set(HAVE_QT5 ON)
-      set(HAVE_QT  ON)
-      find_package(Qt5 COMPONENTS OpenGL QUIET)
-      if(Qt5OpenGL_FOUND)
-        set(QT_QTOPENGL_FOUND ON)
+    find_package(QT NAMES Qt6 Qt5 COMPONENTS Core Gui Widgets Test Concurrent REQUIRED NO_MODULE)
+    find_package(Qt${QT_VERSION_MAJOR} COMPONENTS Core Gui Widgets Test Concurrent REQUIRED NO_MODULE)
```

Comment:
Cases with:
- `WITH_QT=4`
- `WITH_QT=5`
- `WITH_QT=6`
- and `WITH_QT=ON`

should be handled properly.

Reply:
This block is not used to handle QT4, 
please see "if(NOT WITH_QT EQUAL 4)" and code at line 42 ...

Reply:
Clean 3.4 branch + Fedora 33 with Qt4 + Qt5:

-  &#x2714; WITH_QT=ON => `YES (ver 5.15.2)`
-  &#x2714; WITH_QT=4 => `YES (ver 4.8.7 EDITION = OpenSource)`
-  &#x2714; WITH_QT=5 => `YES (ver 5.15.2)`
-  &#x2716; WITH_QT=6 => `YES (ver 5.15.2)`

Patch original (May) + Fedora 33 with Qt4 + Qt5:

-  &#x2714; WITH_QT=ON => `YES (ver 5.15.2)`
-  &#x2714; WITH_QT=4 => `YES (ver 4.8.7 EDITION = OpenSource)`
-  &#x2714; WITH_QT=5 => `YES (ver 5.15.2)`
-  &#x2716; WITH_QT=6 => `YES (ver 5.15.2)`

Patch updated + Fedora 33 with Qt4 + Qt5:

-  &#x2714; WITH_QT=ON => `YES (ver 5.15.2)`
-  &#x2714; WITH_QT=4 => `YES (ver 4.8.7 EDITION = OpenSource)`
-  &#x2714; WITH_QT=5 => `YES (ver 5.15.2)`
-  &#x2714; WITH_QT=6 => **FAILED**


---

Need someone with Qt6 environment to validate behavior